test/integration/integration_admin_test.cc

```diff
@@ -442,4 +444,78 @@ TEST_F(IntegrationAdminIpv4Ipv6Test, Ipv4Ipv6Listen) {
   }
 }
 
+// Testing the behavior of StatsMatcher, which allows/denies the  instantiation of stats based on
+// restrictions on their names.
+class StatsMatcherIntegrationTest : public HttpIntegrationTest, public testing::Test {
+public:
+  StatsMatcherIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, Network::Address::IpVersion::v4,
+                            realTime()) {}
```

Comment:
IMO this test is appropriate for simTime() rather than realTime(). Can you try that?